src/allora_sdk/worker/worker.py

```diff
@@ -117,63 +146,72 @@ def inferer(
-    #         api_key: API key for testnet faucet (if needed)
-    #         topic_id: The Allora network topic ID to submit predictions to
-    #         fee_tier: Transaction fee tier (ECO/STANDARD/PRIORITY)
-    #         polling_interval: Interval in seconds to poll for new submission windows
-    #         debug: Enable debug logging
-
-    #     Returns:
-    #         An instance of AlloraWorker configured as a reputer
-    #     """
-    #     return cls(
-    #         run=run,
-    #         run_pkl=run_pkl,
-    #         wallet=wallet,
-    #         network=network,
-    #         api_key=api_key,
-    #         topic_id=topic_id,
-    #         fee_tier=fee_tier,
-    #         polling_interval=polling_interval,
-    #         submission_window_event_type=EventReputerSubmissionWindowOpened,
-    #         debug=debug,
-    #     )
+    @classmethod
+    def reputer(
+        cls,
+        run: LossFn = default_squared_error_loss,
+        *,
+        ground_truth_fn: GroundTruthFn,
+        wallet: Optional[AlloraWalletConfig] = None,
+        network: AlloraNetworkConfig = AlloraNetworkConfig.testnet(),
+        api_key: Optional[str] = None,
+        topic_id: int = 69,
+        fee_tier: FeeTier = FeeTier.STANDARD,
+        polling_interval: int = 120,
+        min_stake_uallo: Optional[int] = None,
+        debug: bool = False,
+    ):
+        """
+        Create an AlloraWorker configured as a reputer.
+
+        Args:
+            run: Loss function to calculate error between ground truth and predictions
+            ground_truth_fn: Function that returns ground truth values (str or float)
+            wallet: Wallet configuration (private key, mnemonic, or file)
+            network: Allora network configuration (testnet/mainnet/custom)
+            api_key: API key for testnet faucet (if needed)
+            topic_id: The Allora network topic ID to submit reputer payloads to
+            fee_tier: Transaction fee tier (ECO/STANDARD/PRIORITY)
+            polling_interval: Interval in seconds to poll for new submission windows
+            min_stake_uallo: Minimum stake in uallo to top-up to (used for dynamic staking)
+            debug: Enable debug logging
+
+        Returns:
+            An instance of AlloraWorker configured as a reputer
+        """
+        return cls(
+            run=run,
+            ground_truth_fn=ground_truth_fn,
+            wallet=wallet,
+            network=network,
+            api_key=api_key,
+            topic_id=topic_id,
+            fee_tier=fee_tier,
+            polling_interval=polling_interval,
+            submission_window_event_type=EventReputerSubmissionWindowOpened,
+            role=WorkerRole.REPUTER,
+            min_stake_uallo=min_stake_uallo,
+            debug=debug,
+        )
 
 
     def __init__(
         self,
-        run: PredictFn,
+        run: Union[PredictFn, LossFn],
+        ground_truth_fn: Optional[GroundTruthFn] = None,
```

Comment:
I had envisioned all three worker types having only the `run` function (although with different inputs/outputs).  Do you think we need to make `ground_truth_fn` separate?  Or can we just rely on the end user to include this in `run`?  We can make some helper utilities for reputers, but generally speaking I think it's better if we avoid special-casing as much as possible in this part of the code